packages/unified-selection/src/unified-selection/queries/ECMetadata.ts

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/** @packageDocumentation
+ * @module UnifiedSelection
+ */
+
+/**
+ * An interface of an iModel metadata provider used to retrieve information about ECSchemas,
+ * ECClasses, ECProperties, etc.
+ *
+ * @beta
+ */
+export interface IMetadataProvider {
```

Comment:
let's have an internal discussion on how to avoid duplicating all these types

Reply:
Summary of the discussion:
- For the scope of this PR, we keep these types as duplicates here, but make sure they're not exported and don't have release tags.
- In the long term, we'll need to come up with a way to share some code between packages - either through a shared package or shared code that gets transpiled into each package that needs it.